leshan-client-demo/src/main/java/org/eclipse/leshan/client/demo/LeshanClientDemo.java

```diff
@@ -312,20 +321,100 @@ public static void main(final String[] args) {
                 return;
             }
         }
+
         // Set boolean controlling OSCORE usage
-        boolean useOSCore = false;
+        OSCoreSettings oscoreSettings = null;
         if (cl.hasOption("oscore")) {
             System.out.println("Using OSCORE");
-            useOSCore = true;
 
             HashMapCtxDB db = OscoreHandler.getContextDB();
             OSCoreCoapStackFactory.useAsDefault(db);
 
+            // Parse OSCORE related command line parameters
```

Comment:
We should begin to check if configuration is consistent and all oscore mandatory option are present. (like we do for PSK/RPK/X509)

Reply:
Done in commit 0caaf4569fe0d6a380310ad2f9a4d87eb642f86d